### PR TITLE
Simplified _ThreadFuture use of covers.

### PR DESCRIPTION
**Pull request by davidhesselbom**

### CODE REVIEW

**Comment by davidhesselbom** [comment]
Hold on, I missed something...

File: source/base/Promise.ooc [hunk 1]
```diff
@@ -102,10 +102,8 @@ _ThreadFuture: class <T> extends Future<T> {
 			temporary := task()
 			if (T inheritsFrom?(Object))
 				this _result = temporary
-			else {
+			else
 				this _result = Cell<T> new(temporary)
-				this _hasCover = true
```
Comment (marcusnaslund) [inline source/base/Promise.ooc:107]: Delete line 96 as well?

Comment (davidhesselbom) [inline source/base/Promise.ooc:107]: That's the thing I missed, yeah :) Fixed now.

**Comment by davidhesselbom** [comment]
Updated.

**Comment by marcusnaslund** [comment]
Hmm, I believe I might have already done this but lost it when I merged some other changes.
Thanks anyhow. :)

**Pull request merged**